### PR TITLE
added test for POST /begroing/begroing_result

### PR DESCRIPTION
**Pull request by haakondr**
- had to fix a lot of hardcoded unit and variable id's. I am not very
happy about the amount of fetching that needs to be done. We should
consider adding some additional abstraction layers or perhaps write more
complex queries with nested queries? not sure what is best..

TODO: for some reason the mocking did not work and it called the actual implementation.

Another yet awkward thing is that I added aquamonitor credentials for running integration tests, BUT this test did also call aquamonitor to write as my mocking did not work which resulted in a write. I have cleaned up now, but removed the aquamonitor writing as part of test. Need to isolate this better and ideally get a read only user for integration tests.

### CODE REVIEW

**Comment by RmStorm** [comment]
>     * had to fix a lot of hardcoded unit and variable id's. I am not very
>       happy about the amount of fetching that needs to be done. We should
>       consider adding some additional abstraction layers or perhaps write more
>       complex queries with nested queries? not sure what is best..
> 
> 
> TODO: for some reason the mocking did not work and it called the actual implementation.
> 
> Another yet awkward thing is that I added aquamonitor credentials for running integration tests, BUT this test did also call aquamonitor to write as my mocking did not work which resulted in a write. I have cleaned up now, but removed the aquamonitor writing as part of test. Need to isolate this better and ideally get a read only user for integration tests.

A read only user is an excellent plan. Is that easy to do @roarbra?

File: src/odm2_postgres_api/routes/begroing_routes.py [hunk 1]
```diff
@@ -72,17 +77,23 @@ async def post_begroing_result(begroing_result: schemas.BegroingResultCreate,
...
             completed_action = await post_actions(data_action, connection)
             for result_index in method_observations:
+
                 data_result = schemas.ResultsCreate(
                     samplingfeatureuuid=begroing_result.station.samplingfeatureuuid,
                     actionid=completed_action.actionid,
                     resultuuid=str(uuid.uuid4()),
                     resulttypecv=result_type_and_unit_dict[method.methodname][0],
-                    variableid=10,  # This variable indicates the abundance of the taxon of the result
+                    variableid=variable.variableid,  # This variable indicates the abundance of the taxon of the result
```
Comment (RmStorm) [inline src/odm2_postgres_api/routes/begroing_routes.py:93]: The comment can be removed if `variable` is renamed to `abundance_variable`.

File: tests/test_begroing.py [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
-from odm2_postgres_api.routes.shared_routes import post_directive, post_sampling_features
-from odm2_postgres_api.schemas.schemas import DirectivesCreate, SamplingFeaturesCreate, BegroingIndicesCreate
+
+from odm2_postgres_api.routes.begroing_routes import post_indices, post_begroing_result
+from odm2_postgres_api.routes.shared_routes import post_directive, post_sampling_features, post_taxonomic_classifiers
+from odm2_postgres_api.schemas.schemas import DirectivesCreate, SamplingFeaturesCreate, BegroingIndicesCreate, \
+    BegroingResultCreate, Methods, TaxonomicClassifierCreate
+from integration_test_fixtures import db_conn, wait_for_db
+
+user_header = "eyJpZCI6IDIyMSwgInVpZCI6ICIxZWQyMDBkMy1mMDlhLTQxNjQtOTExMC1hMWYyNGY4OTliYjMiLCAiZGlzcGxheU5hbWUiOiAiXHUwMGM1Z2UgT2xzZW4iLCAiZW1haWwiOiAiZGV2dXNlckBzb21lZW1haWwuY29tIiwgInByb3ZpZGVyIjogIkRldkxvZ2luIiwgImNyZWF0ZVRpbWUiOiAiMjAyMC0wNC0yMFQxMTo0NToyMS4yNDFaIiwgInVwZGF0ZVRpbWUiOiAiMjAyMC0wNC0yMFQxMTo0NToyMS4yNDFaIiwgInJvbGVzIjogWyJhcHBzOmFkbWluIiwgIm5pdmEiXX0="
```
Comment (RmStorm) [inline tests/test_begroing.py:14]: Can we completely capitalize global hardcoded variables like these? like so: `USER_HEADER`

**Review by RmStorm — CHANGES_REQUESTED** [review]
I think there is some inconsistencies in your local copy of ODM2, you should wipe and rerun it. Do you want to wait with merging this until you have refactored the frontend? I'm also ok with merging it directly after fixing the errors.

**Comment by haakondr** [comment]
made issue about creating read only aquamonitor API user: https://github.com/NIVANorge/AquaMonitor/issues/22

File: src/odm2_postgres_api/routes/begroing_routes.py [hunk 3]
```diff
@@ -43,21 +42,27 @@ async def post_begroing_result(begroing_result: schemas.BegroingResultCreate,
...
         used_method_indices = [i for i, e in enumerate(begroing_result.observations[index]) if e]
         if len(used_method_indices) != 1:
             raise ValueError('Must have one and only one method per species')
         observations_per_method[used_method_indices[0]].append(index)
 
-    # 13 is 'Microscopic semi quantitative abundance classifier', 17 is a percentage
+    unit_micr_abundance = await find_unit(connection, UnitsCreate(unitstypecv="Dimensionless", unitsabbreviation="-",
+                                                                  unitsname="Presence or Absence"), raise_if_none=True)
+    unit_macro_coverage = await find_unit(connection, UnitsCreate(unitstypecv="Dimensionless", unitsabbreviation="%",
+                                                                  unitsname="Kiselalger Relative abundance"),
```
[the remark this answers is not in the document]
Comment (haakondr) [reply src/odm2_postgres_api/routes/begroing_routes.py]: you are right. Rebuilt now and it looks as you describe.

Food for thought: we risk this happening quite a bit, and by describing units/variables etc as database rows, we need a good regime for keeping everyone up to date.

**Comment by roarbra** [comment]
> 
> A read only user is an excellent plan. Is that easy to do @roarbra?

@RmStorm That could be solved by an Internal user, for instance your own user. Such a user will have full read access, but write access only to those projects that he has editing rights for. It's a little bit more job to introduce a test user, because we have a job that wipes out all Internal usernames that isn't within Active Directory.

**Pull request merged**